go/cmd/server/logging/logging.go

```diff
@@ -0,0 +1,306 @@
+		for idx, featureName := range fcoSchema.Features {
+			// populate the proto value arrays with values from memory buffer in separate columns one for each feature name
+			if _, ok := columnNameToProtoValueArray[featureName]; !ok {
+				columnNameToProtoValueArray[featureName] = make([]*types.Value, 0)
+				columnNameToStatus[featureName] = make([]int32, 0)
+				columnNameToTimestamp[featureName] = make([]int64, 0)
+			}
+			columnNameToProtoValueArray[featureName] = append(columnNameToProtoValueArray[featureName], l.FeatureValues[idx])
+			columnNameToStatus[featureName] = append(columnNameToStatus[featureName], int32(l.FeatureStatuses[idx]))
+			columnNameToTimestamp[featureName] = append(columnNameToTimestamp[featureName], l.EventTimestamps[idx].AsTime().UnixNano()/int64(time.Millisecond))
+		}
+	}
+
+	fields := make([]arrow.Field, 0)
+	columns := make([]array.Interface, 0)
+	for _, entityName := range fcoSchema.Entities {
+		protoArr := entityNameToEntityValues[entityName]
+		if len(protoArr) == 0 {
+			break
+		}
+		valArrowArray, err := gotypes.ProtoValuesToArrowArray(protoArr, arrowMemory, len(columnNameToProtoValueArray))
+		if err != nil {
+			return nil, err
+		}
+		arrowType, err := gotypes.ValueTypeEnumToArrowType(fcoSchema.EntityTypes[entityName])
+		if err != nil {
+			return nil, err
+		}
+		fields = append(fields, arrow.Field{
+			Name: entityName,
+			Type: arrowType,
+		})
+		columns = append(columns, valArrowArray)
+	}
+
+	for _, featureName := range fcoSchema.Features {
+
+		protoArr := columnNameToProtoValueArray[featureName]
+		if len(protoArr) == 0 {
+			break
+		}
+		arrowArray, err := gotypes.ProtoValuesToArrowArray(protoArr, arrowMemory, len(columnNameToProtoValueArray))
+		if err != nil {
+			return nil, err
+		}
+
+		arrowType, err := gotypes.ValueTypeEnumToArrowType(fcoSchema.FeaturesTypes[featureName])
+
+		if err != nil {
+			return nil, err
+		}
+		fields = append(fields, arrow.Field{
+			Name: featureName,
+			Type: arrowType,
+		})
+		columns = append(columns, arrowArray)
+	}
+	schema := arrow.NewSchema(
+		fields,
+		nil,
+	)
+
+	result := array.Record(array.NewRecord(schema, columns, int64(len(memoryBuffer.logs))))
+
+	tbl := array.NewTableFromRecords(schema, []array.Record{result})
+	return array.Table(tbl), nil
+}
+
+type Schema struct {
+	Entities      []string
+	Features      []string
+	EntityTypes   map[string]types.ValueType_Enum
+	FeaturesTypes map[string]types.ValueType_Enum
+}
+
+func GetSchemaFromFeatureService(featureService *model.FeatureService, entities []*model.Entity, featureViews []*model.FeatureView, onDemandFeatureViews []*model.OnDemandFeatureView) (*Schema, error) {
+	fvs := make(map[string]*model.FeatureView)
+	odFvs := make(map[string]*model.OnDemandFeatureView)
+
+	entityNames := make([]string, 0)
+	for _, entity := range entities {
+		entityNames = append(entityNames, entity.JoinKey)
```

Comment:
we need to add only join keys that are required by currect feature service not all entities that exist

Reply:
also please use `joinKey` name

Reply:
In the current data model, I don't think there is a direct connection between entities and feature service or entity names and featureviews.